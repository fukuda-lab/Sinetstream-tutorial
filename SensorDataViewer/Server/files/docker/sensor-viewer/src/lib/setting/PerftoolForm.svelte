<script lang="ts">
  import { ValidationMessage } from "@felte/reporter-svelte";
</script>

<article class="border" aria-labelledby="perftool-group-title">
  <details open>
    <summary class="none">
      <div class="row">
        <h6 id="perftool-group-title">Perftool</h6>
        <div class="max" />
        <i>arrow_drop_down</i>
      </div>
    </summary>

    <ValidationMessage for="perftool.name" let:messages={message}>
      <div class="field label">
        <input type="text" name="perftool.name" id="perftool-name" />
        <label for="perftool-name">perftool name</label>
        <span class="error">{message || ""}</span>
      </div>
    </ValidationMessage>

    <label class="switch icon">
      <input
        type="checkbox"
        name="perftool.visible"
        aria-label="perftool chart switch"
      />
      <span>
        <i>show_chart</i>
      </span>
      <div class="tooltip">perftool chart</div>
    </label>
  </details>
</article>

<style>
  .field.label > label {
    color: silver;
  }
</style>
