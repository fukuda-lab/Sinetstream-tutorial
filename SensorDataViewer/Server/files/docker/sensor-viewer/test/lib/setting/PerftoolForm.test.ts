import { cleanup, render, screen } from '@testing-library/svelte';
import {
  afterEach, describe, expect, it,
} from 'vitest';
import PerftoolForm from '../../../src/lib/setting/PerftoolForm.svelte';

describe('PerftoolForm.svelt', () => {
  afterEach(() => cleanup());

  it('mounts', () => {
    const { container } = render(PerftoolForm);
    expect(container).toBeTruthy();
  });

  it('name', () => {
    render(PerftoolForm);
    const item = screen.getByRole('textbox', { name: 'perftool name' });
    expect(item).toBeTruthy();
    expect(item.parentElement?.outerHTML).toMatchSnapshot();
    expect(item).toHaveProperty('name', 'perftool.name');
  });

  it('switch', () => {
    render(PerftoolForm);
    const item = screen.getByRole('checkbox', { name: 'perftool chart switch' });
    expect(item).toBeTruthy();
    expect(item.parentElement?.outerHTML).toMatchSnapshot();
    expect(item).toHaveProperty('name', 'perftool.visible');
  });

  it('expansion open', () => {
    render(PerftoolForm);
    const item = screen.getByRole('group');
    expect(item.getAttribute('open')).not.toBeNull();
  });
});
