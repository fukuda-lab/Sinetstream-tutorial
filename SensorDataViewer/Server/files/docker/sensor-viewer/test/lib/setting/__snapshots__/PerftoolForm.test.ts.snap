// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PerftoolForm.svelt > name 1`] = `"<div class=\\"field label s-mPlSbL43WKAe\\"><input type=\\"text\\" name=\\"perftool.name\\" id=\\"perftool-name\\" class=\\"s-mPlSbL43WKAe\\"> <label for=\\"perftool-name\\" class=\\"s-mPlSbL43WKAe\\">perftool name</label> <span class=\\"error s-mPlSbL43WKAe\\"></span></div>"`;

exports[`PerftoolForm.svelt > switch 1`] = `"<label class=\\"switch icon s-mPlSbL43WKAe\\"><input type=\\"checkbox\\" name=\\"perftool.visible\\" aria-label=\\"perftool chart switch\\" class=\\"s-mPlSbL43WKAe\\"> <span class=\\"s-mPlSbL43WKAe\\"><i class=\\"s-mPlSbL43WKAe\\">show_chart</i></span> <div class=\\"tooltip s-mPlSbL43WKAe\\">perftool chart</div></label>"`;
